### PR TITLE
create additional role Sidecar Manager, reviewed existing permissions

### LINKED ISSUE #12044: GET /api/sidecars/all requires admin privileges. Should it ?
I'm building an internal dashboard and need data reported by `/api/sidecars/all` Naturally I wanted to use a user with minimal permisions for this API endpoint. By trial and error (couldn't find a definitive list of roles and privileges required by api endpoints) I realised that only the "Admin" role grants enough permisions to read this endpoint.

## Expected Behavior

I expected that the reader role should be enough to do `GET` on `/api/sidecars/all` The next obvious (to me) role to try was "Sidecar System (Internal)" but again no luck. I added roles one by one and retried but the only role that granted me access was "admin"

## Current Behavior

I get 
```json
{
  "type": "ApiError",
  "message": "Not authorized"
}
```
with any available role other then "Admin" indiferent if I use a token or user/pass authentication.

## Possible Solution

- add a "sidecar reader" role
- grant the "Sidecar System (Internal)" role permisions to this endpoint

## Steps to Reproduce (for bugs)

1. Create a user. Assign all the roles but admin - including the one you would expect to work: Sidecar System (Internal)
2. Use that user to GET `/api/sidecars/all`
3. Fail

## Context

I'm a strong believer in using the minimum required permissions to accomplish a task. In case of dashboards, status and monitoring I always want "reader users" i.e. users that can get data and status but can't perform changes. Currently having  admin credentials sitting on the dashboard system is not ideal. 

## Your Environment

* Graylog Version: 4.2.5+59802bf (official docker)
* Java Version:  1.8.0_312
* Elasticsearch Version: 7.10.2 (official docker)
* MongoDB Version: latest 4.2 (official docker)
* Operating System: official docker image
* Browser version: Firefox 96.0.2

### PR DESCRIPTION
**Pull request by moesterheld**
## Description
reviewed permissions of Sidecar System user, made `sidecars/configurations` unavailable to the user since it is only used from the UI, not the sidecar.
added Sidecar Manager role with permissions to manage and read sidecars

## Motivation and Context
closes #12044 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Review by AntonEbel — CHANGES_REQUESTED** [review]

File: changelog/unreleased/pr-15380.toml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+type = "a"
+message = "Add Sidecar Manager role to read and manage sidecars."
```
Comment (AntonEbel) [inline changelog/unreleased/pr-15380.toml:2]: Perhaps we should also mention here the role of the `Sidecar Reader`.

File: graylog2-server/src/main/java/org/graylog/plugins/sidecar/migrations/V20230502164900_AddSidecarManagerRole.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+import com.google.common.collect.ImmutableSet;
+import org.graylog.plugins.sidecar.permissions.SidecarRestPermissions;
+import org.graylog2.migrations.Migration;
+import org.graylog2.migrations.MigrationHelpers;
+
+import javax.inject.Inject;
+import java.time.ZonedDateTime;
+
+public class V20230502164900_AddSidecarManagerRole extends Migration {
```
Comment (AntonEbel) [inline graylog2-server/src/main/java/org/graylog/plugins/sidecar/migrations/V20230502164900_AddSidecarManagerRole.java:27]: Maybe `V20230502164900_AddSidecarManagerAndReaderRole`?

**Review by AntonEbel — CHANGES_REQUESTED** [review]
Just some minor things and one thing for the `Sidecar Reader` role. Currently, this user can click on the edit and update buttons, which results in an error page. I think it would improve the UX if someone from the frontend team would hide these buttons for the reader role. I don't know if we should make these changes in this PR or create an issue for that.

**Review by AntonEbel — APPROVED** [review]
LGTM

**Review by mpfz0r — APPROVED** [review]
LGTM as well :+1:

**Comment by ryan-carroll-graylog** [comment]
@kroepke @moesterheld @AntonEbel can this be backported to 5.0?

There's a HubSpot issue open for another customer with this same issue (HS-1821909158) who's on Graylog 5.0.10.

**Pull request merged**